integration/spark/src/main/scala/org/apache/spark/sql/CarbonScan.scala

```diff
@@ -151,8 +153,15 @@ case class CarbonScan(
 
         override def next(): InternalRow = {
           val value = iter.next
+          val converted = value.map {
```

Comment:
There was performance problem earlier with map if number of scan rows are more, that was the main reason we moved it to carbon layer.